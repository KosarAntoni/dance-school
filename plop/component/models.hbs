export type {{properCase name}}Props {

}