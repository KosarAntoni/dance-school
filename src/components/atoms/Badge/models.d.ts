export type BadgeProps = {
  variant?: 'arrow' | 'nono' | 'translucent' | string;
  color?: 'green' | 'blue' | 'orange' | 'white' | string;
  className?: string;
};

export type Badge = {
  label: string;
} & BadgeProps;
