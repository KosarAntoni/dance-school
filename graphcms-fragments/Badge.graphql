fragment Badge on Badge {
  stage
  locale
  remoteId: id
  createdAt(variation: COMBINED)
  updatedAt(variation: COMBINED)
  publishedAt(variation: COMBINED)
  label
  createdBy {
    ... on User {
      remoteTypeName: __typename
      remoteId: id
      stage
    }
  }
  updatedBy {
    ... on User {
      remoteTypeName: __typename
      remoteId: id
      stage
    }
  }
  publishedBy {
    ... on User {
      remoteTypeName: __typename
      remoteId: id
      stage
    }
  }
  variant
  color
  scheduledIn {
    ... on ScheduledOperation {
      remoteTypeName: __typename
      remoteId: id
      stage
    }
  }
}