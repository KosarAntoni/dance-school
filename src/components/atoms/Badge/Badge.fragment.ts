import { graphql } from 'gatsby';

export const ButtonFragment = graphql`
  fragment Badge on GraphCMS_Badge {
    color
    id
    label
    variant
  }
`;
