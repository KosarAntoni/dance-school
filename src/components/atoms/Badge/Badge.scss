@use 'sass:map';
@import 'styles/component-set';

.badge {
  position: relative;
  z-index: map.get($z-index, 'ground');
  display: inline-flex;
  align-items: center;
  border-radius: 4px;

  &::before {
    position: absolute;
    left: 0;
    z-index: map.get($z-index, 'underground');
    width: 100%;
    height: 100%;
    border-radius: inherit;
    content: '';
  }

  &--translucent {
    &::before {
      opacity: 0.3;
    }
  }

  &--arrow {
    margin-left: 8px;

    &::after {
      position: absolute;
      left: -6px;
      z-index: map.get($z-index, 'underground');
      height: 74%;
      border-radius: inherit;
      content: '';
      transform: rotate(45deg);
      aspect-ratio: 1 / 1;
      justify-self: flex-end;
    }
  }

  &--orange {
    &::before,
    &::after {
      background-color: $c-orange;
    }

    &.badge {
      &--translucent {
        color: $c-orange;
      }
    }
  }

  &--blue {
    color: $c-white;

    &::before,
    &::after {
      background-color: $c-pastel-blue;
    }

    &.badge {
      &--translucent {
        color: $c-pastel-blue;
      }
    }
  }

  &--green {
    &::before,
    &::after {
      background-color: $c-green;
    }

    &.badge {
      &--translucent {
        color: $c-green;
      }
    }
  }

  &--white {
    &::before,
    &::after {
      background-color: $c-white;
    }

    &.badge {
      &--translucent {
        color: $c-white;
      }
    }
  }
}
