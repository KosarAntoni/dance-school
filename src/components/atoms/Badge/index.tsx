export { default } from './Badge';
export * from './models.d';
