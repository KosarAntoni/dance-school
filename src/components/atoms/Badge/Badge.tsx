import React, { FC } from 'react';
import classNames from 'classnames';

import Typography from 'atoms/Typography';

import { BadgeProps } from './models.d';

import './Badge.scss';

const Badge: FC<BadgeProps> = ({ children, variant = 'nano', color = 'white', className }) => {
  const badgeClasses = classNames(
    'badge',
    {
      [`badge--${variant}`]: variant,
      [`badge--${color}`]: color,
    },
    className
  );

  return (
    <Typography
      className={badgeClasses}
      size={14}
      padding={{ top: 'xxs', right: 'xs', bottom: 'xxs', left: 'xs' }}
      data-testid="badge-item"
    >
      {children}
    </Typography>
  );
};

export default Badge;
