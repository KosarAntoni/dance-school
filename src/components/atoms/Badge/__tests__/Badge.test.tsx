import React from 'react';
import { render, screen } from '@testing-library/react';

import {
  BadgeChildrenMock,
  BadgeDefaultClassesMock,
  BadgeDefaultPropsMock,
} from '../__mock__/mock';
import Badge from '../Badge';

import { BadgeProps } from '../models';

const defaultProps: BadgeProps = {
  ...BadgeDefaultPropsMock,
};

test('the Badge component renders correctly with required and optional props', () => {
  render(<Badge {...defaultProps}>{BadgeChildrenMock}</Badge>);

  const badgeNode = screen.getByTestId('badge-item');

  // the component has been rendered
  expect(badgeNode).toBeInTheDocument();

  // generating a snapshot, not for testing purposes but only to see what changes in the HTML structure during CR if the developer updates the snapshot
  expect(badgeNode).toMatchSnapshot();

  // the comonent has correct classes
  expect(badgeNode).toHaveClass(BadgeDefaultClassesMock.join(' '));
});
