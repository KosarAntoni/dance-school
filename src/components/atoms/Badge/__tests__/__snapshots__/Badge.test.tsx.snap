// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Badge component renders correctly with required and optional props 1`] = `
<div
  class="typography typography--size-14 typography--padding-top-xxs typography--padding-right-xs typography--padding-bottom-xxs typography--padding-left-xs typography--line-height-normal badge badge--translucent badge--orange test-class"
  data-testid="badge-item"
>
  test content
</div>
`;
